src/sas/qtgui/Plotting/SlicerParameters.py

```diff
@@ -142,58 +187,163 @@ def onFocus(self, row, column):
-            if plot not in self.active_plots:
-                continue
-            # get the plotter2D instance
-            plotter = self.active_plots[plot]
-            # Assign model to slicer
-            index = self.cbSlicer.currentIndex()
-            slicer = self.callbacks[index]
-            plotter.setSlicer(slicer=slicer)
-            # override slicer model
-            plotter.slicer._model = self.model
-            # force conversion model->parameters in slicer
-            plotter.slicer.setParamsFromModel()
-        pass
+            # Apply plotter to a plot
+            self.applyPlotter(plot)
+            # Save 1D plots if required
+            plots.append(plot)
+        if self.isSave and self.model is not None:
+            self.save1DPlotsForPlot(plots)
+        pass  # debug anchor
+
+    def applyPlotter(self, plot):
+        """
+        Apply the current slicer to a plot
+        """
+        # don't assign to itself
+        if plot == self.parent.data[0].name:
+            return
+        # a plot might have been deleted
+        if plot not in self.active_plots:
+            return
+        # get the plotter2D instance
+        plotter = self.active_plots[plot]
+        # Assign model to slicer
+        index = self.cbSlicer.currentIndex()
+
+        slicer = self.callbacks[index]
+        if slicer is None:
+            plotter.onClearSlicer()
+            return
+        plotter.setSlicer(slicer=slicer, reset=False)
+        # override slicer model
+        plotter.slicer._model = self.model
+        # force conversion model->parameters in slicer
+        plotter.slicer.setParamsFromModel()
+
+    def save1DPlotsForPlot(self, plots):
+        """
+        Save currently shown 1D sliced data plots for a given 2D plot
+        """
+        items_for_fit = []
+        for plot in plots:
+            for item in self.active_plots.keys():
+                data = self.active_plots[item].data[0]
+                if not isinstance(data, Data1D):
+                    continue
+                if plot not in data.name:
+                    continue
+                filename = data.name if self.txtExtension.text() == ""\
+                    else data.name + "_" + str(self.txtExtension.text())
+                # Saved as "TXT" file, as in original impl
+                extension = self.cbSaveExt.currentText()
+                filename_ext = filename + extension
+                filepath = os.path.join(self.save_location, filename_ext)
+                if 'txt' in extension:
+                    GuiUtils.onTXTSave(data, filepath)
+                elif 'xml' in extension:
+                    loader = Loader()
+                    loader.save(filepath, data, ".xml")
+                elif 'h5' in extension:
+                    nxcansaswriter = NXcanSASWriter()
+                    nxcansaswriter.write([data], filepath)
+                new_item = GuiUtils.createModelItemWithPlot(data, name=filename)
```

Comment:
New DataExplorer objects are not appended with '[n]' so Data1D objects can end up with the same display name (same slicer, same data file, switching back and forth between slicers).

Reply:
Yes, I noticed that just recently - will fix it

Reply:
Absolutely.
Fixed.